include/ck/utility/debug.hpp

```diff
@@ -4,6 +4,15 @@
 #ifndef UTILITY_DEBUG_HPP
 #define UTILITY_DEBUG_HPP
 
+#ifdef ENABLE_CK_DEBUG
+#define CK_DEBUG(stmts) \
+    {                   \
+        stmts           \
+    }
```

Comment:
Why not making it more feature rich ? ;) I know right now its con is it's simplicity, however to make it easier to write nice log messages by developers I'd add here printing message mechanism and logging file and line at the same time. I imagine the use would look similar to this:
```
CK_DEBUG("Your message starts here. You can print some value") <<  variable_a;
```
When it would return a stream it would be easy to write more compound messages.

I'd even think of using Google's [glog](https://github.com/google/glog#getting-started) library which is very nice!

Reply:
> Why not making it more feature rich ? ;) I know right now its con is it's simplicity, however to make it easier to write nice log messages by developers I'd add here printing message mechanism and logging file and line at the same time. I imagine the use would look similar to this:
> 
> ```
> CK_DEBUG("Your message starts here. You can print some value") <<  variable_a;
> ```
> 
> When it would return a stream it would be easy to write more compound messages.
> 
> I'd even think of using Google's [glog](https://github.com/google/glog#getting-started) library which is very nice!

I think by writing code this way, we are adding overhead in production (even some functions are conditional empty).

Reply:
I thought that if "disabled" such logging libraries shouldn't add any overhead to the code.